### PR TITLE
drivers/mtd_sdcard : Expose Configurations to Kconfig 

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in MTD SD Card Storage driver to Kconfig. 

#### Important 

 ~~MTD_SDCARD_SKIP_ERASE to be deprecated and CONFIG_MTD_SDCARD_ERASE introduced.~~

MTD_SDCARD_SKIP_ERASE removed and CONFIG_MTD_SDCARD_ERASE introduced, since the feature is not implemented yet.

### Testing procedure

Doxygen build works fine.

Following Macro was introduced in main.c for testing.
```
#define STR(x)   #x
#define SHOW_DEFINE(x) printf("%s=%s\n", #x, STR(x))
```
USEMODULE += mtd_sdcard introduced in make file.

##### Firmware Output

RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2020.07-devel-174-g79e16-Kconfig_mtd_sdcard_tests)
CONFIG_MTD_SDCARD_ERASE=CONFIG_MTD_SDCARD_ERASE
MTD_SDCARD_SKIP_ERASE=(1)

#### Usage with CFLAGS 

New File -> /tests/driver_mtd_sdcard/Makefile

> CFLAGS += -DCONFIG_MTD_SDCARD_ERASE=1

##### Firmware Output

RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2020.07-devel-174-g79e16-Kconfig_mtd_sdcard_tests)
CONFIG_MTD_SDCARD_ERASE=1
MTD_SDCARD_SKIP_ERASE=(0)

#### Usage with Kconfig

New Folder  -> /tests/driver_mtd_sdcard

> make menuconfig

##### Firmware Output

RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2020.07-devel-174-g79e16-Kconfig_mtd_sdcard_tests)
CONFIG_MTD_SDCARD_ERASE=1
MTD_SDCARD_SKIP_ERASE=(0)

Note : Hardware is not available fro interfacing hence configurability of macros were only tested.

### Issues/PRs references

#12888
@leandrolanzieri

### CODE REVIEW

File: drivers/Kconfig.net [hunk 1]
```diff
@@ -21,3 +21,7 @@ rsource "ads101x/Kconfig"
 rsource "hdc1000/Kconfig"
 rsource "mma8x5x/Kconfig"
 endmenu # Sensor Device Drivers
+
+menu "Storage Device Drivers"
```
Comment (leandrolanzieri) [inline drivers/Kconfig.net:25]: Please move to `Kconfig`

Comment (akshaim) [reply drivers/Kconfig.net]: Done.

File: drivers/include/mtd_sdcard.h [hunk 2]
```diff
@@ -48,18 +49,37 @@ typedef struct {
...
+ *          the sector first.
+ *          Attention: an erase call will therefore NOT touch the content,
+ *          so disable this feature to ensure overriding the data.
+ *
+ *          @deprecated Use inverse @ref CONFIG_MTD_SDCARD_ERASE instead.
+ *          Will be removed after 2021.01 release.
  */
 #ifndef MTD_SDCARD_SKIP_ERASE
+#if IS_ACTIVE(CONFIG_MTD_SDCARD_ERASE)
+#define MTD_SDCARD_SKIP_ERASE (0)
```
Comment (leandrolanzieri) [inline drivers/include/mtd_sdcard.h:78]: On an offline chat @MichelRottleuthner indicated that this functionality is actually not supported yet. So I would say it is safe to change the macro logic without deprecating this. Let's just replace the macro

Comment (akshaim) [reply drivers/include/mtd_sdcard.h:78]: Okay. I shall update

Comment (akshaim) [reply drivers/include/mtd_sdcard.h]: Done.

**Pull request merged**